### PR TITLE
feat: add concurrency and queue limits to OCR service

### PR DESCRIPTION
**Pull request by xirf**
- Add `maxConcurrency` and `maxQueueSize` to `PaddleOptions` to limit simultaneous and queued `recognize()` and `deskewImage()` operations.
- Implement a `Semaphore` in `BasePaddleOcrService` to gate operations and enforce FIFO ordering.
- Reject new requests immediately if the queue exceeds `maxQueueSize`.
- Update memory management by copying ONNX tensor data before disposal and cleaning up canvases after use.
- Add tests for queue overflow and concurrency limits.
- Update README with new configuration options and Kaggle dataset setup.

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -48,4 +48,6 @@ export const DEFAULT_PADDLE_OPTIONS: PaddleOptions = {
   recognition: DEFAULT_RECOGNITION_OPTIONS,
   debugging: DEFAULT_DEBUGGING_OPTIONS,
   session: DEFAULT_SESSION_OPTIONS,
+  maxConcurrency: 1,
```
Comment (snowfluke) [inline src/constants.ts]: these two here is anti-pattern

**Review by snowfluke — CHANGES_REQUESTED** [review]
